src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -402,6 +402,7 @@ private void LowerConstantValueDecision(DecisionTree.ByValue byValue)
                     {
                         var loweredRight = kv.Key;
                         var loweredLeft = kv.Value;
+                        loweredRight = _factory.Convert(loweredLeft.Type, loweredRight);
```

Comment:
I see an invalid IL generated for this code:
```
class Test
{
void M(MyStruct s)
{
switch (s)
{
	case I1 o:
		System.Console.WriteLine(o);
		break;
}
}
}

interface I1{}

struct MyStruct : I1 {}
```

So it doesn't look like we have existing tests covering all of the mentioned scenarios. Please expand the testing to conversions that aren't limited to conversions to object.